app/main.py

```diff
@@ -1 +1,62 @@
-#  write your code here
+from __future__ import annotations
+
+
+class Animal:
+    def __init__(
+        self,
+        name: str,
+        appetite: int,
+        is_hungry: bool = True
+    ) -> Animal:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = not self.is_hungry
+            return self.appetite
+
+        return 0
+
+
+class Cat(Animal):
+    def __init__(
+        self,
+        name: str,
+        appetite: int = 3,
+        is_hungry: bool = True
+    ) -> Cat:
+        super().__init__(name, appetite, is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(
+        self,
+        name: str,
+        appetite: int = 7,
+        is_hungry: bool = True
+    ) -> Dog:
+        super().__init__(name, appetite, is_hungry)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list) -> int:
+    feed_result = 0
+    for animal in animals:
+        if animal.is_hungry:
+            feed_result += animal.appetite
+            animal.feed()
```

Comment:
You can simplify this method using sum() function , comprehension and taking into account that feed returns self.appetite and checks if the animal is hungry

Reply:
Done, code has become more simplify, as for me - it become more difficult to understand..